@neil-hickey awesome, thanks!

I've added the test.

It fails on the develop branch:

```
--- FAIL: TestEmptyLineStartsConfig (0.00s)
    --- FAIL: TestEmptyLineStartsConfig/whitespace_in_config_is_ignored (0.00s)
        config_test.go:27:
                Error Trace:    /home/zebradil/development/forks/vendir/pkg/vendir/config/config_test.go:27
                Error:          Received unexpected error:
                                Parsing resource config '/tmp/TestEmptyLineStartsConfigwhitespace_in_config_is_ignored185744596/001/config.yml': Unknown apiVersion '' or kind '' for resource
                Test:           TestEmptyLineStartsConfig/whitespace_in_config_is_ignored
FAIL
FAIL    command-line-arguments  0.004s
FAIL
```

It passes on the feature branch:

```
ok      command-line-arguments  0.004s
```

--- a/pkg/vendir/config/config_test.go
+++ b/pkg/vendir/config/config_test.go
@@ -1,0 +1,29 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package config_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+)
+
+func TestEmptyLineStartsConfig(t *testing.T) {
+	t.Run("Empty document is ignored", func(t *testing.T) {
+		tempConfigPath := filepath.Join(t.TempDir(), "config.yml")
+		configWithWhitespace := []byte(`
+
+---
+apiVersion: vendir.k14s.io/v1alpha1
+kind: Config`)
+
+		require.NoError(t, os.WriteFile(tempConfigPath, configWithWhitespace, 0666))
+
+		_, _, _, err := config.NewConfigFromFiles([]string{tempConfigPath})
+		require.NoError(t, err)
+	})
+}

